PR for CA JS2

sulenchy opened the pull request.

Miksel90 committed:
big update

Miksel90 committed:
Update index.html

Miksel90 committed:
fixing dropdowns

Miksel90 committed:
Update index.html

Miksel90 committed:
compledted the feed page

Miksel90 committed:
fixed name in feed

Miksel90 committed:
fixed navigation

Miksel90 committed:
fixed icons on small screens

Miksel90 committed:
Update index.scss

Miksel90 committed:
fixed the background image on hero page

Miksel90 committed:
fiex the background to remove inline styling

Miksel90 committed:
styling

Miksel90 committed:
Update README.md

Miksel90 committed:
Create register.js

Miksel90 committed:
fixed button

Miksel90 committed:
working on the register code

Miksel90 committed:
update register

fixed register js. user can register succsesfully

Miksel90 committed:
login

created login. js

Miksel90 committed:
Update login.js

successfully stored accestoken and redirected to profile page if respone = true

Miksel90 committed:
Merge pull request #2 from Miksel90/login-development

Login development

Miksel90 committed:
moved files

Miksel90 committed:
Started on profile changes

started on the profile Get request.
Need to filter it down to the person logging in somehow.

Miksel90 committed:
Update profileInfo.js

Miksel90 committed:
made the profile join over to the profile page

Miksel90 committed:
updated profile page

dynamically build profile info on page from the user API.

Follow button not in use right now.

Miksel90 committed:
trying to create the all posts by logged in user function

Miksel90 committed:
Update index.html

Miksel90 committed:
fixed issue

Miksel90 committed:
update

Miksel90 committed:
Update profileposts.js

Miksel90 committed:
updated profile posts Js

fetching Profile posts and info
added Fontawesome link to all pages

Miksel90 committed:
fixed icons

Miksel90 committed:
Merge pull request #3 from Miksel90/profile-development

Profile development

Miksel90 committed:
updated the fetching all posts js

building html from the fetch and styling

Miksel90 committed:
workin gon filtering

Miksel90 committed:
contnued stylin gon the post

fixed images and like icon

Miksel90 committed:
added some styleing

Miksel90 committed:
working on the search function

Miksel90 committed:
working on search function

Miksel90 committed:
updateing code

Miksel90 committed:
filter posts

manages to filter posts by date and if reaction=true sort by number of reactions.

having some issues with bootstrap dropdown button so had to create two buttons instead...going back to that when i can

Miksel90 committed:
Update filterFeedPosts.js

Miksel90 committed:
working on modal

Miksel90 committed:
building my posts specfic

workin gon adding a comment section

Miksel90 committed:
built new post specific page

gave up modal idea. built post specific page built up by id from post. Trying to implement a back to top btn. but sturggelign with bootstrap

Miksel90 committed:
fixed scrollbutton

Miksel90 committed:
fixing scroll

Miksel90 committed:
adding new images and building post specific page

Miksel90 committed:
buliding up my post section

Miksel90 committed:
feed developing

working on design on my feed page.
User able to create post and site reloads after posting. site is very slow =(

Miksel90 committed:
modal update

manged to dynamically build a button that opens a modal with the id from that post. Now i need to implement a edit and a delte code for the buttons.

Miksel90 committed:
working on the put code

Miksel90 committed:
working on my edit button

Miksel90 committed:
working on editing

Miksel90 committed:
trying to get my modal edit to work

Miksel90 committed:
got my edit and delete buttons working

Miksel90 committed:
Merge pull request #4 from Miksel90/feed-development

Feed development

Miksel90 committed:
Update README.md

Miksel90 committed:
dadsa

Miksel90 committed:
Update .gitignore

Miksel90 committed:
Update register.js

Miksel90 committed:
fixed filtering

fixed the filtering
now i need to add a by default button

Miksel90 committed:
trying to chain more functions

Miksel90 committed:
trying to cchain fucntions

Miksel90 committed:
fixing modules

Miksel90 committed:
cleaning

Miksel90 committed:
Merge pull request #5 from Miksel90/feed-development

Feed development

Miksel90 committed:
Update populateFeed.js

sulenchy commented:
@Miksel90 As far as the UI and the functionality are concerned, you have the following intact

- Login 
- register
- fetch all posts
- fetch a single post
- Search

And you have the following yet to be implemented:

- filter
- delete post
- update post

However, I noticed a bug where an auth user can access the login page. This should not be the case. Only unauthenticated user should access register and login page. And solution is to write a function that check if the user is auth or not and act accordingly

sulenchy commented on js/backToTopButton.js:
```diff
@@ -0,0 +1,17 @@
+window.onscroll = () => {
```
This looks good to me

sulenchy commented on js/deletePost.js:
```diff
@@ -0,0 +1,44 @@
+const modal = document.getElementById("myModal");
+const deleteButton = document.getElementById("deletePostButton");
+
+async function deletePost(postId) {
```
You have delete function? Ok. I can't seems to see that in the UI

Miksel90 replied:
if you click the edit button on the profilepage posts its open a modal where you can edit and delete the post

sulenchy commented on js/deletePost.js:
```diff
@@ -0,0 +1,44 @@
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${token}`,
+      },
+      body: JSON.stringify({
```
According to the API doc., this is incorrect. The id is meant to be passed to the endpoint in the url lik `/social/posts/<id>`

sulenchy reviewed (commented):
Good job! so far. Please, try and complete the rest of the functionality ASAP.

Miksel90 committed:
comments succesfull

user can comment to other uses posts!

Miksel90 committed:
Update index.html

Miksel90 committed:
edit profile image

added logged in users to be able to change profile image!

Miksel90 committed:
changed href

Miksel90 committed:
Update index.html

Miksel90 committed:
reactions

User can react to other posts. There was no way to remove reactions unfortunatly.

Miksel90 committed:
styling

Miksel90 committed:
working on "friends" section

Miksel90 committed:
working on follow/unfollow

Miksel90 committed:
style

Wave checked everything!
added "friends" to profile page
working on follow/unfollow from feed page.
Thinking of just simplyfing it a lot

Miksel90 committed:
wave check post specific

Miksel90 committed:
resolving issues

Miksel90 committed:
Update .gitignore

Miksel90 committed:
Update .gitignore

Miksel90 committed:
Update index.html

Miksel90 committed:
Merge branch 'Js-2' into Optional-features-

Miksel90 committed:
Merge pull request #7 from Miksel90/Optional-features-

Optional features

Miksel90 committed:
Update const.mjs

Miksel90 committed:
jsdocs

Added jsdocs to all functions and put some consts in another file and exported/imported

Miksel90 committed:
styling
